libraries/ortb2Utils/currency.js

```diff
@@ -0,0 +1,15 @@
+const additionalData = new WeakMap();
+
+export function getCurrencyFromBidderRequest(bidderRequest) {
+  return (bidderRequest?.ortb2?.ext?.cur || [])[0];
+}
+
+export function setAdditionalData(obj, key, value) {
```

Comment:
if these are only used for adserverCurrency, they'd be better as `set/getAdServerCurrency` (they are in an currency library after all, and unlikely to be used for anything else).

But if they are only used for one adapter, I don't think they're needed at all - the adapter itself can maintain the map.